### PR TITLE
refactor: update scaffolding declarative Copilot

### PR DESCRIPTION
**Pull request by yuqizhou77**
- update scaffolding based on latest decision.
  - rename "GPT" to "declarative copilot".
  -  We will provide two options for user to start with, a basic declarative copilot or a declarative copilot with a plugin from on Azure function

New scaffold flow:
- basic Declarative Copilot
image
- Declarative Copilot with Plugin from scratch
image

TODO: will add a CLI command ("teamsapp add plugin" for example) so that user can add plugin/action from an existing API spec to a Declarative Copilot project.

### CODE REVIEW

File: packages/fx-core/src/question/create.ts [hunk 1]
```diff
@@ -921,7 +922,7 @@ export function capabilityQuestion(): SingleSelectQuestion {
         case ProjectTypeOptions.customCopilot().id:
           return getLocalizedString("core.createProjectQuestion.projectType.customCopilot.title");
         case ProjectTypeOptions.customizeGpt().id:
-          return "Choose the GPT type";
+          return "Choose GPT type";
```
Comment (huimiu) [inline packages/fx-core/src/question/create.ts:925]: Do we also need to replace this GPT with declarative copilot?

File: packages/fx-core/src/question/create.ts [hunk 2]
```diff
@@ -864,17 +864,18 @@ export class CapabilityOptions {
   // customize GPT
   static customizeGptBasic(): OptionItem {
     return {
-      id: "customize-gpt-basic",
-      label: "Basic GPT",
-      detail: "A simple GPT skeleton you can author without any plugin",
+      id: "basic-declarative-copilot",
+      label: "Basic Declarative Copilot",
```
Comment (tecton) [inline packages/fx-core/src/question/create.ts:868]: Please remember to do localization.

**Review by jayzhang — APPROVED** [review]

**Review by tecton — APPROVED** [review]

File: packages/fx-core/src/question/create.ts (hunk 2 above)
Comment (yuqizhou77) [reply packages/fx-core/src/question/create.ts:868]: sure! going to do that later. Currently this option is for CLI only.

**Review by huimiu — DISMISSED** [review]

File: packages/fx-core/src/question/create.ts (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/src/question/create.ts]: Thanks, updated.

**Review by huimiu — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**